### PR TITLE
rocSOLVER support 3

### PR DESCRIPTION
**Pull request by tfalders**
Added getri.

### CODE REVIEW

**Comment by tfalders** [comment]
PR is ready for review now. getri bug should be fixed on rocSOLVER side.

File: library/src/hcc_detail/hipblas.cpp [hunk 1]
```diff
@@ -12320,6 +12320,50 @@ rocblas_status rocsolver_zgeqrf_ptr_batched(rocblas_handle                handle
                                             rocblas_double_complex* const ipiv[],
                                             const rocblas_int             batch_count);
 
+rocblas_status rocsolver_sgetri_outofplace_batched(rocblas_handle       handle,
```
Comment (daineAMD) [inline library/src/hcc_detail/hipblas.cpp:12323]: Doesn't look like this is being exposed, is it for future use?

File: library/src/hcc_detail/hipblas.cpp [hunk 2]
```diff
@@ -13031,6 +13075,183 @@ hipblasStatus_t hipblasZgetrsStridedBatched(hipblasHandle_t          handle,
                                          batch_count));
 }
 
+// getri
+hipblasStatus_t hipblasSgetri(
```
[the remark this answers is not in the document]
Comment (tfalders) [reply library/src/hcc_detail/hipblas.cpp:13079]: This is one of those annoying cases again where the cuBLAS API differs from the rocSOLVER API. Juan wants getri to do an in-place computation as in vanilla LAPACK, whereas cuBLAS getri_batched does an out-of-place computation and puts the result in C. I created the outofplace_batched method to match this signature, which is not exposed in our public API and is only for use in hipBLAS, but I didn't create outofplace methods for the normal and strided_batched routines.

File: clients/include/testing_getri_batched.hpp [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+                else
+                    hA[b][i + j * lda] -= 4;
+            }
+        }
+
+        // perform LU factorization on A
+        hInfo[b] = cblas_getrf(M, N, hA[b].data(), lda, hIpivb);
+
+        // Copy data from CPU to device
+        CHECK_HIP_ERROR(hipMemcpy(bA[b], hA[b].data(), A_size * sizeof(T), hipMemcpyHostToDevice));
```
[the remark this answers is not in the document]
Comment (tfalders) [reply clients/include/testing_getri_batched.hpp:99]: C is a pure output parameter. Anything copied into bC[b] will just be ignored and overridden by the getri call.

**Pull request merged**